### PR TITLE
Fix the property list workaround for ruby ~2.0

### LINKED ISSUE #11199: [Regression] pilot upload fails in 2.69.0: private method 'include' called for Array:Class (NoMethodError)
### New Issue Checklist

- [✔] Updated fastlane to the latest version
- [✔] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [✔] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [✔] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
I submit app builds to TestFlight through Visual Studio Team Services continuous integration, which uses Fastlane. The CI system is set up to automatically update Fastlane if an update is available for each release. That's how I found this regression.

Yesterday, the release worked fine. Checking the logs, I found that testflight version `2.68.2` was used then. Today's release failed (it had updated TestFlight to `2.69.0`).  It reliably fails each time on this new version.

Note: in the meanwhile, I have configured the CI system to force Fastlane version `2.68.2` for now to work around this issue.

Here's the command the CI system runs, and the failing output, when using Fastlane `2.69.0`.

##### Complete output when running fastlane, including the stack trace and command used
`fastlane pilot upload -u ******** -i /Users/*******/buildagent/_work/r5/a/*******/ios_release/******.ipa -a com.*****.******* --skip_waiting_for_build_processing true`

```
fastlane pilot upload -u ******** -i /Users/*******/buildagent/_work/r5/a/*******/ios_release/******.ipa -a com.*****.******* --skip_waiting_for_build_processing true
2017-12-12T23:42:13.4643610Z /Users/xamarin/.gem-cache/gems/fastlane-2.69.0/fastlane_core/lib/fastlane_core/core_ext/cfpropertylist.rb:17:in `block in <top (required)>': private method `include' called for Array:Class (NoMethodError)
2017-12-12T23:42:13.4743770Z 	from /Users/xamarin/.gem-cache/gems/fastlane-2.69.0/fastlane_core/lib/fastlane_core/core_ext/cfpropertylist.rb:14:in `each'
2017-12-12T23:42:13.4846830Z 	from /Users/xamarin/.gem-cache/gems/fastlane-2.69.0/fastlane_core/lib/fastlane_core/core_ext/cfpropertylist.rb:14:in `<top (required)>'
2017-12-12T23:42:13.4945890Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-12-12T23:42:13.5052070Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-12-12T23:42:13.5149200Z 	from /Users/xamarin/.gem-cache/gems/fastlane-2.69.0/fastlane_core/lib/fastlane_core/ipa_file_analyser.rb:2:in `<top (required)>'
2017-12-12T23:42:13.5179650Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-12-12T23:42:13.5209300Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-12-12T23:42:13.5236760Z 	from /Users/xamarin/.gem-cache/gems/fastlane-2.69.0/fastlane_core/lib/fastlane_core.rb:19:in `<top (required)>'
2017-12-12T23:42:13.5266310Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-12-12T23:42:13.5295180Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-12-12T23:42:13.5325740Z 	from /Users/xamarin/.gem-cache/gems/fastlane-2.69.0/fastlane/lib/fastlane.rb:1:in `<top (required)>'
2017-12-12T23:42:13.5357100Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-12-12T23:42:13.5396980Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-12-12T23:42:13.5427300Z 	from /Users/xamarin/.gem-cache/gems/fastlane-2.69.0/fastlane/lib/fastlane/cli_tools_distributor.rb:18:in `take_off'
2017-12-12T23:42:13.5457120Z 	from /Users/xamarin/.gem-cache/gems/fastlane-2.69.0/bin/fastlane:20:in `<top (required)>'
2017-12-12T23:42:13.5484290Z 	from /usr/local/bin/fastlane:23:in `load'
2017-12-12T23:42:13.5510660Z 	from /usr/local/bin/fastlane:23:in `<main>'
2017-12-12T23:42:13.5714150Z ##[error]Error: fastlane failed with return code: 1
```

### Environment

(I can't easily run this command since Fastlane is run through a continuous integration system. Let me know if you need this and I'll figure out a way.)

### PR DESCRIPTION
**Pull request by milch**
Fixes #11199

### CODE REVIEW

File: fastlane_core/lib/fastlane_core/core_ext/cfpropertylist.rb [hunk 1]
```diff
@@ -14,5 +14,5 @@
 [Array, Hash].each do |c|
   c.send :alias_method, :to_binary_plist, :to_plist
   c.send :remove_method, :to_plist
-  c.include Plist::Emit
+  c.module_eval "include Plist::Emit"
```
Comment (jdee) [inline fastlane_core/lib/fastlane_core/core_ext/cfpropertylist.rb]: Good work. This is, of course, entirely equivalent to:

```Ruby
class Array
  alias_method :to_binary_plist, :to_plist
  remove_method :to_plist
  include Plist::Emit
end

class Hash
  alias_method :to_binary_plist, :to_plist
  remove_method :to_plist
  include Plist::Emit
end
```

At first I had Enumerator in the list as well, so this saved a lot. We could just expand it out, which could be simpler.

**Review by jdee — APPROVED** [review]
:+1: Is this in response to the issue that was reported, or something else?

**Comment by jdee** [comment]
Another thought: As of RuboCop 0.50, Ruby 2.0 is no longer supported. There was a `TargetRubyVersion` added to .rubocop.yml recently that breaks as soon as you upgrade RuboCop.

File: fastlane_core/lib/fastlane_core/core_ext/cfpropertylist.rb (hunk 1 above)
Comment (milch) [reply fastlane_core/lib/fastlane_core/core_ext/cfpropertylist.rb]: I tried expanding it out like that but it didn't work (on 2.0.0), so I left it like that. I think this is a pretty short piece of code, so it should be fine

**Comment by milch** [comment]
@jdee Yes, this is in response to the issue reporting the regression #11199. 

Honestly I think soon it'll be time to formally drop support for 2.0.0. System ruby on High Sierra is already at 2.3 and most CI services I know of use later versions anyway.

**Pull request merged**